Add support for Ubuntu 24.04

matthewandrews opened the pull request.

vjrj commented on ansible/roles/common/tasks/setfacts.yml:
```diff
@@ -177,6 +177,41 @@
+    pg_version: "{{ pg_version | default('16') }}"
+    postgis_version: "{{ postgis_version | default('3') }}"
+    postgis_version_folder: "{{ postgis_version_folder | default('3') }}"
+    postgis_scripts_version: "{{ postgis_scripts_version | default('3.4') }}"
```
I think that this should be `3`:

```
$ apt search postgis | grep scripts

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

postgresql-16-pgrouting-scripts/noble 3.6.1-1 all
  Routing functionality support for PostgreSQL/PostGIS - SQL scripts
postgresql-16-postgis-3-scripts/noble 3.4.2+dfsg-1ubuntu3 all
  Geographic objects support for PostgreSQL 16 -- SQL scripts
postgresql-pgrouting-scripts/noble 3.6.1-1 all
  Routing functionality support for PostgreSQL/PostGIS - SQL scripts metapackage
postgresql-postgis-scripts/noble 3.4.2+dfsg-1ubuntu3 all
  Geographic objects support for PostgreSQL -- SQL scripts metapackage
```

matthewandrews replied:
You mean `postgis_scripts_version` default should be `3` ?  It seems like the relevant version is `3.4.2` for `noble`...

vjrj commented on ansible/roles/augeas/tasks/main.yml:
```diff
@@ -99,3 +99,17 @@
+    - augeas-tools
+    - python3-augeas
+    - python3-libxml2
+    - libpython3.10
```
```
E: Unable to locate package libpython3.10
```

matthewandrews replied:
Thanks, fixed in https://github.com/AtlasOfLivingAustralia/ala-install/pull/833/commits/e1d1b37b2643ab22458a72c18ad50a0de50851e3

vjrj commented on ansible/group_vars/all/vars.yml:
```diff
@@ -1 +1 @@
-supported_ansible_version: {full: 2.10.3, major: 2, minor: 10, revision: 3}
+supported_ansible_version: {full: 2.17.3, major: 2, minor: 17, revision: 3}
```
Related: https://github.com/AtlasOfLivingAustralia/ala-install/pull/800

matthewandrews replied:
Thanks, I've merged #800 into `main` and into this branch now.

vjrj reviewed (changes requested):
In general LGTM except for the minor comments I added, and the fact that I couldn't test correctly without  merge https://github.com/AtlasOfLivingAustralia/ala-install/pull/800. 

I get errors like:

```
NoduleNotFoundError: No module named 'ansible.module_utils.six.moves
```
with older `ansible` like the recommended by current `ala-install` or:

```
ansible.builtin.include has been removed. Use include_tasks or import_tasks instead. This feature was removed from ansible-core in a release after 2023-05-16. Please update your playbooks.
```
with recent `ansible` and `ala-install` without #800.

vjrj commented:
Something like this can be also added for make newserver to work: 
```
diff --git a/ansible/newserver.yml b/ansible/newserver.yml
index afa3f810..e6b79d10 100644
--- a/ansible/newserver.yml
+++ b/ansible/newserver.yml
@@ -1,15 +1,30 @@
 ---
 # Use this playbook once on a new server to install python2 and python3
 # must pass target var e.g. ansible-playbook newserver.yml -i /path/to/inventories/your-inventory-file -u ${USER} --private-key=~/.ssh/id_rsa --extra-vars 'target=bla-bla'
+
 - name: newserver
   hosts: '{{ target }}'
   gather_facts: false
+  become: yes
+  
   pre_tasks:
-  - name: Install python2 and python3 for Ansible
+  - name: Get the distribution version
+    raw: lsb_release -r | awk '{print $2}'
+    register: os_version
+    changed_when: false
+
+  - name: Install python2 and python3 for Ansible (< u24)
     raw: bash -c "test -e /usr/bin/python || (apt -qqy update && apt-get install -qqy python3-minimal 'python2.*-minimal')"
-    register: output
-    changed_when: output.stdout != ""
+    when: os_version.stdout is version('24.04', '<')
+    register: output_python2_and_3
+    changed_when: output_python2_and_3.stdout != ""
+  
+  - name: Install only python3 for Ansible (>= u24)
+    raw: bash -c "test -e /usr/bin/python || (apt -qqy update && apt-get install -qqy python3-minimal)"
+    when: os_version.stdout is version('24.04', '>=')
+    register: output_python3
+    changed_when: output_python3.stdout != ""
+
+  tasks:
   - name: Gathering Facts
     setup:
-  become: yes
-
```

matthewandrews commented:
> Something like this can be also added for make newserver to work:
> 
> ```
> diff --git a/ansible/newserver.yml b/ansible/newserver.yml
> index afa3f810..e6b79d10 100644
> --- a/ansible/newserver.yml
> +++ b/ansible/newserver.yml
> @@ -1,15 +1,30 @@
>  ---
>  # Use this playbook once on a new server to install python2 and python3
>  # must pass target var e.g. ansible-playbook newserver.yml -i /path/to/inventories/your-inventory-file -u ${USER} --private-key=~/.ssh/id_rsa --extra-vars 'target=bla-bla'
> +
>  - name: newserver
>    hosts: '{{ target }}'
>    gather_facts: false
> +  become: yes
> +  
>    pre_tasks:
> -  - name: Install python2 and python3 for Ansible
> +  - name: Get the distribution version
> +    raw: lsb_release -r | awk '{print $2}'
> +    register: os_version
> +    changed_when: false
> +
> +  - name: Install python2 and python3 for Ansible (< u24)
>      raw: bash -c "test -e /usr/bin/python || (apt -qqy update && apt-get install -qqy python3-minimal 'python2.*-minimal')"
> -    register: output
> -    changed_when: output.stdout != ""
> +    when: os_version.stdout is version('24.04', '<')
> +    register: output_python2_and_3
> +    changed_when: output_python2_and_3.stdout != ""
> +  
> +  - name: Install only python3 for Ansible (>= u24)
> +    raw: bash -c "test -e /usr/bin/python || (apt -qqy update && apt-get install -qqy python3-minimal)"
> +    when: os_version.stdout is version('24.04', '>=')
> +    register: output_python3
> +    changed_when: output_python3.stdout != ""
> +
> +  tasks:
>    - name: Gathering Facts
>      setup:
> -  become: yes
> -
> ```

Thanks! I've added that change in https://github.com/AtlasOfLivingAustralia/ala-install/pull/833/commits/d57c3c781625adc62d3c81f334b0939242559349

vjrj commented:
LGTM and can be merged IMHO. I continue adding other issues I detect in #834.
